src/js/hooks/cycleCount/useAllProductsTab.jsx

```diff
@@ -135,15 +135,23 @@ const useAllProductsTab = ({ filterParams }) => {
           {translate('react.cycleCount.table.binLocation.label', 'Bin Location')}
         </TableHeaderCell>
       ),
-      cell: ({ getValue }) => (
-        <TableCell
-          className="rt-td"
-          tooltip
-          tooltipLabel={getValue()}
-        >
-          {mapStringToList(getValue(), ',', 100).map((binLocationName) => <div>{binLocationName}</div>)}
-        </TableCell>
-      ),
+      cell: ({ getValue }) => {
+        const binLocationList = mapStringToList(getValue(), ',');
+
+        return (
+          <TableCell
+            className="rt-td"
+            tooltip
+            tooltipLabel={`${getValue()} (${binLocationList.length})`}
+          >
+            {binLocationList.map((binLocationName) => (
+              <div>
+                {binLocationName.length > 16 ? `${binLocationName.substring(0, 16)}...` : binLocationName}
+              </div>
+            ))}
+          </TableCell>
+        );
```

Comment:
I think we should do this by CSS, not just limiting the length, because when you make the screen even smaller (I know we are supporting certain screen sizes, but a lot of users have smaller screens than those supported) the bin location won't be visible. You can achieve that by packing every bin location to the div and then just hiding the overflow.